### PR TITLE
Use image previews instead of full-resolution

### PR DESCRIPTION
**Pull request by ajacksified**
Don't merge until we get the all-clear from @xiongchiamiov.

:eyeglasses: @jo-asakura or @danehansen

### CODE REVIEW

File: src/views/components/Listing.jsx [hunk 1]
```diff
@@ -54,6 +54,18 @@ class Listing extends React.Component {
     this.state = {
       expanded: this.props.expanded,
     };
+
+    if (typeof window !== 'undefined') {
```
Comment (jo-asakura) [inline src/views/components/Listing.jsx]: Would just `if (window)` work?

File: src/views/components/Listing.jsx [hunk 2]
```diff
@@ -107,6 +119,42 @@ class Listing extends React.Component {
     );
   }
 
+  previewImageUrl(listing, expanded) {
```
Comment (jo-asakura) [inline src/views/components/Listing.jsx]: Just a thought.. would it be nice to have a safe-guard for nullable listing? If listing is `null` some things below will error out.

**Comment by jo-asakura** [comment]
:fish:

File: src/views/components/Listing.jsx (hunk 2 above)
Comment (ajacksified) [inline src/views/components/Listing.jsx]: good call - I'll update.

File: src/views/components/Listing.jsx (hunk 1 above)
Comment (ajacksified) [inline src/views/components/Listing.jsx]: Nope, throws on the server side.

**Pull request merged**